client/my-sites/site-settings/podcasting-details/link.jsx

```diff
@@ -113,20 +117,23 @@ class PodcastingLink extends Component {
 export default connect( ( state, ownProps ) => {
 	const { fields } = ownProps;
 
-	const podcastingCategoryId = Number( fields && fields.podcasting_category_id );
-	const isPodcastingEnabled = podcastingCategoryId > 0;
-
 	const siteId = getSelectedSiteId( state );
 	const siteSlug = getSelectedSiteSlug( state );
 
+	const podcastingCategoryId = Number( fields && fields.podcasting_category_id );
+	const podcastingCategory =
+		podcastingCategoryId > 0 && getTerm( state, siteId, 'category', podcastingCategoryId );
+
```

Comment:
I actually think the `getPodcastingCategory` selector approach is better. Easier to read this code (a long `connect`/any function seems like a code smell), and all the data is already in the state tree. Is there something I'm missing? Perhaps related to handling the case on the details screen where the category has been changed but not yet saved?

Reply:
As far as the `connect` function, there would be one line of difference.

With a selector:

```js
const podcastingCategory = getPodcastingCategory( state, siteId, podcastingCategoryId );
```

Without a selector (current approach):

```js
const podcastingCategoryId = Number( fields && fields.podcasting_category_id );
const podcastingCategory =
	podcastingCategoryId > 0 && getTerm( state, siteId, 'category', podcastingCategoryId );
```

In order to add the new selector, we need to add about 100 lines of code + tests, and it is only used in this one place to save a single line of code.  IMO not worth it.

> Perhaps related to handling the case on the details screen where the category has been changed but not yet saved?

This is also worth thinking about.  Since we are already on a separate screen from the details screen at this point, hopefully there are no issues with the settings `fields` being out of sync with what is actually saved on the site on this point.  I haven't seen anything like this so far.

Reply:
Thanks for explaining the reasoning for the non-selector approach further. I agree with your conclusion. 👍